### PR TITLE
use fuse.js for search  #840

### PR DESCRIPTION
**Pull request by wjhrdy**
**Summary**
I needed a fuzzy search for my own use case so I thought I'd add the feature. It is probably better suited as a plugin.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:
before
before

after
after

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

---

* [ ] DO NOT include files inside `lib` directory.

### CODE REVIEW

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -88,75 +88,70 @@ export function search(query) {
...
+    minMatchCharLength: 1,
+    keys: [{
+      name: 'title',
+      weight: 0.75
+    }, {
+      name: 'body',
+      weight: 0.25
+    }]
+  };
+  var fuse = new Fuse(data, options);
```
Comment (jthegedus) [inline src/plugins/search/search.js]: where's Fuse coming from?

Comment (wjhrdy) [reply src/plugins/search/search.js]: gotta include it as in index as a separate dependency.

cdn here:
https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.4.4/fuse.min.js

documentation:
https://fusejs.io/

Comment (jthegedus) [reply src/plugins/search/search.js]: could you please update the docs to reflect this within this PR?

**Comment by anikethsaha** [comment]
cc @wjhrdy whats the status on this?

Also, can we make this search as opt-in which can be configured.

**Pull request closed without merging**